Brucheion/Brucheion
# Self-updating capabilities

### Functionality this PR provides:

* Add self-updating capabilities to Brucheion: users can obtain and install the newest release of Brucheion directly rather than downloading it themselves.
* Self-updating capabilities are not enabled by default and require the `-update` flag
* Users are prompted to confirm the installation; the application will run as usual otherwise

### How to test:

* Create kind-of-a release build by setting the `main.Version` property to something less than the most recent release, e.g. `go build -v -ldflags "-X main.Version=v2.1.0 -X main.BuildTime=$(date +%FT%T%z)"`
* Run the resulting binary with the `-update` flag: `./Brucheion -update`
* Confirm the prompt to install the available update. The previously built binary will be replaced by the "newer" one of the release obtained from GitHub.
* Verify the installation by running the binary

### Please note:

* Self-updating does not work with development builds. By setting the above mentioned `main.Version` property of the application, we can provide a distinct version to builds, similar to what our release pipeline does (Git tag → `main.Version`). Development builds don't have such properties by default, and hence updating would not work properly.
* By default, the updating mechanism does not use pre-releases (as stated on the respective GitHub release) but will use the highest semver version

### To-do

- [x] Build releases as tar-gzipped binaries
- [x] Add self-updating capabilities to Brucheion (behind an `-update` flag?)
- [x] ~Add visual out-of-date indicators to UI (with hints on how to update)~ (will raise another issue/PR)
- [x] Fix self-updating on Windows 10
